### ISSUE TITLE
Update Security Group for Trunk ENI when ENIConfig is updated

### ISSUE THREAD

**Issue by jdn5126** [aws/amazon-vpc-resource-controller-k8s#373]
**What would you like to be enhanced**:
When Custom Networking and Security Groups for Pods are configured, the VPC Resource Controller creates the trunk interface with the Security Group ID specified in the `ENIConfig`. If that `ENIConfig` is updated, the trunk SGID is not updated. This enhancement request is to update the trunk ENI SGID.

**Why is the change needed and what use case will it solve**:
This change will improve the customer experience by removing the need to terminate and/or create new nodes after making changes to the `ENIConfig` (when Security Groups for Pods is in use).

**Comment by haouc** [comment]
I don't think we have a plan to modify existing network interfaces. Since the resources is referenced by ID, updating rules should be preferred. If have to change security groups or subnets, nodes should be replaced. Closing this issue for now.
